# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/04/01 03:35:17-05:00 len.brown@intel.com 
#   [ACPI] delete extraneous IRQ->pin mappings below IRQ 16
#   http://bugzilla.kernel.org/show_bug.cgi?id=2408
# 
# arch/x86_64/kernel/io_apic.c
#   2004/04/01 03:33:04-05:00 len.brown@intel.com +5 -1
#   don't add extraneous IRQ-> pin mappings when _PRT reside < IRQ 16
# 
# arch/i386/kernel/io_apic.c
#   2004/04/01 03:33:04-05:00 len.brown@intel.com +5 -1
#   don't add extraneous IRQ-> pin mappings when _PRT reside < IRQ 16
# 
diff -Nru a/arch/i386/kernel/io_apic.c b/arch/i386/kernel/io_apic.c
--- a/arch/i386/kernel/io_apic.c	Thu Apr  1 03:35:25 2004
+++ b/arch/i386/kernel/io_apic.c	Thu Apr  1 03:35:25 2004
@@ -2440,7 +2440,11 @@
 	entry.polarity = active_high_low;
 	entry.mask  = 1;
 
-	add_pin_to_irq(irq, ioapic, pin);
+	/*
+	 * IRQs < 16 are already in the irq_2_pin[] map
+	 */
+	if (irq >= 16)
+		add_pin_to_irq(irq, ioapic, pin);
 
 	entry.vector = assign_irq_vector(irq);
 
diff -Nru a/arch/x86_64/kernel/io_apic.c b/arch/x86_64/kernel/io_apic.c
--- a/arch/x86_64/kernel/io_apic.c	Thu Apr  1 03:35:25 2004
+++ b/arch/x86_64/kernel/io_apic.c	Thu Apr  1 03:35:25 2004
@@ -1891,7 +1891,11 @@
 	entry.polarity = active_high_low;
 	entry.mask = 1;					 /* Disabled (masked) */
 
-	add_pin_to_irq(irq, ioapic, pin);
+	/*
+	 * IRQs < 16 are already in the irq_2_pin[] map
+	 */
+	if (irq >= 16)
+		add_pin_to_irq(irq, ioapic, pin);
 
 	entry.vector = assign_irq_vector(irq);
 
